it.baeyens.arduino.common/src/it/baeyens/arduino/common/Common.java

```diff
@@ -254,21 +254,20 @@ public static void StartSerialMonitor(String mComPort) {
     }
 
     public static String[] listComPorts() {
-	Vector<String> SerialList = Serial.list();
-	String outgoing[] = new String[SerialList.size()];
-	SerialList.copyInto(outgoing);
+	List<String> serialList = Serial.list();
+	String[] outgoing = new String[serialList.size()];
+	serialList.toArray(outgoing);
 	return outgoing;
     }
 
     public static String[] listBaudRates() {
-	String outgoing[] = { "115200", "57600", "38400", "31250", "28800", "19200", "14400", "9600", "4800", "2400", //$NON-NLS-1$//$NON-NLS-2$//$NON-NLS-3$//$NON-NLS-4$//$NON-NLS-5$//$NON-NLS-6$//$NON-NLS-7$ //$NON-NLS-8$ //$NON-NLS-9$ //$NON-NLS-10$
-		"1200", "300" }; //$NON-NLS-1$ //$NON-NLS-2$
-	return outgoing;
+	 //$NON-NLS-1$//$NON-NLS-2$//$NON-NLS-3$//$NON-NLS-4$//$NON-NLS-5$//$NON-NLS-6$//$NON-NLS-7$ //$NON-NLS-8$ //$NON-NLS-9$ //$NON-NLS-10$ //$NON-NLS-1$ //$NON-NLS-2$
+	return new String[] {"115200", "57600", "38400", "31250", "28800", "19200", "14400", "9600", "4800", "2400", "1200", "300"};
```

Comment:
I don't think this is an improvement,t as the NON-NLS are to avoid warnings and by having them on the previous line they will add warnings instead of remove warnings